src/katgpucbf/xbgpu/main.py

```diff
@@ -51,56 +51,119 @@
+    """
 
-    - name
-    - dst
+    pass
+
+
+class XOutputDict(OutputDict, total=False):
+    """Configuration options for a baseline-correlation output.
+
+    See :class:`OutputDict` for further information.
+    """
+
+    heap_accumulation_threshold: int
+
+
+def _parse_stream(value: str, kws: _OD, field_callback: Callable[[_OD, str, str], None]) -> None:
+    """Parse a correlation-product or beam stream description.
+
+    This populates `kws` (which should initially be empty) from key=value pairs
+    in `value`. It handles the common fields directly, and type-specific fields
+    are handled by a provided field callback. The callback is invoked with
+    `kws`, the key and the value. If it does not recognise the key, it should
+    raise ValueError.
     """
-    kws: BOutputDict = {}
     for part in value.split(","):
         match part.split("=", 1):
             case [key, data]:
                 match key:
                     case _ if key in kws:
-                        raise ValueError(f"--beam: {key} specified twice")
+                        raise ValueError(f"{key} specified twice")
                     case "name":
                         kws[key] = data
-                    case "channels_per_substream":
-                        kws[key] = int(data)
                     case "dst":
                         kws[key] = endpoint_parser(DEFAULT_PORT)(data)
                     case _:
-                        raise ValueError(f"--beam: unknown key {key}")
+                        field_callback(kws, key, data)
             case _:
-                raise ValueError(f"--beam: missing '=' in {part}")
+                raise ValueError(f"missing '=' in {part}")
     for key in ["name", "dst"]:
-        # These are the bare minimum needed for a beam,
-        # the rest can be taken from X-engine cmd-line args.
         if key not in kws:
-            raise ValueError(f"--beam: {key} is missing")
+            raise ValueError(f"{key} is missing")
+
+
+def parse_beam(value: str) -> BOutput:
+    """Parse a string with beam configuration data.
+
+    The string has a comma-separated list of key=value pairs. See
+    :class:`BOutput` for the valid keys and types. The following
+    keys are required:
+
+    - name
+    - dst
+    """
+
+    def _field_callback(kws: BOutputDict, key: str, data: str) -> None:
+        raise ValueError(f"unknown key {key}")
+
+    try:
+        kws: BOutputDict = {}
+        _parse_stream(value, kws, _field_callback)
+        return BOutput(**kws)
+    except ValueError as exc:
+        raise ValueError(f"--beam: {exc}") from exc
+
 
-    return kws
+def parse_corrprod(value: str) -> XOutput:
+    """Parse a string with correlation product configuration data.
+
+    The string has comma-separated list of key=value pairs. See
+    :class:`XOutput` for the valid keys and types. The following keys
+    are required:
+
+    - name
+    - dst
+    - heap_accumulation_threshold
```

Comment:
Doesn't seem to be required (there is a default), although see my earlier comment suggesting that we make it required.